src/test/java/seedu/address/logic/commands/DeleteAttendanceCommandTest.java

```diff
@@ -0,0 +1,191 @@
+        Map<UUID, Attendance> updatedRecord = new HashMap<>(record);
+        // delete week 1 attendance record
+        updatedRecord.remove(student.getUuid());
+        AttendanceRecord updatedAttendanceRecord = new AttendanceRecord(updatedRecord);
+        List<AttendanceRecord> updatedAttendanceRecords =
+                new ArrayList<>(lesson.getAttendanceRecordList().getAttendanceRecordList());
+        updatedAttendanceRecords.set(targetWeek.getZeroBasedWeekIndex(), updatedAttendanceRecord);
+        Lesson modifiedLesson = new LessonBuilder(lesson)
+                .withAttendanceRecordList(new AttendanceRecordList(updatedAttendanceRecords)).build();
+        ModuleClass modifiedModuleClass =
+                ModuleClassUtil.manualReplaceLessonToModuleClass(moduleClass, lesson, modifiedLesson);
+
+        String expectedMessage = String.format(
+                DeleteAttendanceCommand.MESSAGE_DELETE_ATTENDANCE_SUCCESS, targetWeek,
+                student.getName(), modifiedLesson);
+        Model expectedModel = new ModelManager(model.getTutorsPet(), new UserPrefs());
+        expectedModel.setModuleClass(moduleClass, modifiedModuleClass);
+        expectedModel.commit(expectedMessage);
+
+        DeleteAttendanceCommand deleteAttendanceCommand =
+                new DeleteAttendanceCommand(moduleClassIndex, lessonIndex, studentIndex, targetWeek);
+
+        assertCommandSuccess(deleteAttendanceCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_missingAttendance_failure() {
+        Index moduleClassIndex = INDEX_FIRST_ITEM;
+        Index lessonIndex = INDEX_FIRST_ITEM;
+        Index studentIndex = INDEX_FIRST_ITEM;
+        Week targetWeek = VALID_WEEK_5;
+
+        ModuleClass moduleClass = model.getFilteredModuleClassList().get(moduleClassIndex.getZeroBased());
+        Student student = model.getFilteredStudentList().get(studentIndex.getZeroBased());
+        Lesson lesson = moduleClass.getLessons().get(lessonIndex.getZeroBased());
+        Map<UUID, Attendance> record = lesson.getAttendanceRecordList()
+                .getAttendanceRecord(targetWeek).getAttendanceRecord();
+        Map<UUID, Attendance> updatedRecord = new HashMap<>(record);
+        // delete non-existent week 5 attendance record
+        updatedRecord.remove(student.getUuid());
+        AttendanceRecord updatedAttendanceRecord = new AttendanceRecord(updatedRecord);
+        List<AttendanceRecord> updatedAttendanceRecords =
+                new ArrayList<>(lesson.getAttendanceRecordList().getAttendanceRecordList());
+        updatedAttendanceRecords.set(targetWeek.getZeroBasedWeekIndex(), updatedAttendanceRecord);
+        Lesson modifiedLesson = new LessonBuilder(lesson)
+                .withAttendanceRecordList(new AttendanceRecordList(updatedAttendanceRecords)).build();
+        ModuleClass modifiedModuleClass =
+                ModuleClassUtil.manualReplaceLessonToModuleClass(moduleClass, lesson, modifiedLesson);
```

Comment:
Should the methods from `LessonModificationUtil` and `ModuleClassModificationUtil` be used here instead? Perhaps we can do it after unit tests for these 2 classes are completed.

Reply:
We can use manualReplaceLessonToModuleClass from ModuleClassUtil for now since there are no test cases for those methods which you've mentioned yet